fail2ban/ipaddr.py

```diff
@@ -163,132 +190,108 @@ def __radd__(self, other):
 			domain of the according IP address family
 		"""
-		if self.family == socket.AF_INET:
-			reversed_ip = ".".join(reversed(self.ntoa().split(".")))
+		if self.isIPv4:
+			reversed_ip = self.ntoa.split(".")
 			if not suffix:
-				suffix = "in-addr.arpa."
-
-			return "%s.%s" % (reversed_ip, suffix)
-
-		elif self.family == socket.AF_INET6:
-			reversed_ip = ".".join(reversed(self.hexdump()))
+				suffix = "in-_addr.arpa."
+		elif self.isIPv6:
+			reversed_ip = self.hexdump()
 			if not suffix:
-				suffix =  "ip6.arpa."
-
-			return "%s.%s" % (reversed_ip, suffix)
-			
+				suffix = "ip6.arpa."
 		else:
 			return ""
 
+		return "%s.%s" % (".".join(reversed(reversed_ip)), suffix)
+
+	@property
 	def isIPv4(self):
-		""" return true if the IP object is of address family AF_INET
+		"""Either the IP object is of address family AF_INET
 		"""
 		return self.family == socket.AF_INET
 
+	@property
 	def isIPv6(self):
-		""" return true if the IP object is of address family AF_INET6
+		"""Either the IP object is of address family AF_INET6
 		"""
 		return self.family == socket.AF_INET6
 
-	def getRaw(self):
-		""" returns the raw attribute - should only be set
-			to a non-empty string if prior address conversion
-			wasn't possible
-		"""
-		return self.raw
-
-	def isValidIP(self):
-		""" returns true if the IP object has been created
-			from a valid IP address or false if not
-		"""
-		return self.valid
-
-	
 	@iparg
 	def isInNet(self, net):
-		""" returns true if the IP object is in the provided
-			network (object)
+		"""Return either the IP object is in the provided network
 		"""
-		if self.family != net.family:
+		if self._family != net.family:
 			return False
-
-		if self.family == socket.AF_INET:
+		if self.isIPv4:
 			mask = ~(0xFFFFFFFFL >> net.plen)
-
-		elif self.family == socket.AF_INET6:
+		elif self.isIPv6:
 			mask = ~(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFL >> net.plen)
 		else:
 			return False
 		
-		if self.addr & mask == net.addr:
-			return True
+		return self._addr & mask == net.addr
 
-		return False
-
-		
 	@staticmethod
 	@iparg
 	def masktoplen(mask):
-		""" converts mask string to prefix length
-			only used for IPv4 masks
+		"""Convert mask string to prefix length
+
+		To be used only for IPv4 masks
 		"""
+		mask = mask.addr  # to avoid side-effect within original mask
 		plen = 0
-		while mask.addr:
-			mask.addr = (mask.addr << 1) & 0xFFFFFFFFL
+		while mask:
+			mask = (mask << 1) & 0xFFFFFFFFL
 			plen += 1
 		return plen
 
-
 	@staticmethod
 	def searchIP(text):
-		""" Search if an IP address if directly available and return
-			it.
+		"""Search if text is an IP address, and return it if so, else None
 		"""
 		match = IPAddr.IP_CRE.match(text)
-		if match:
-			return match
-		else:
+		if not match:
 			match = IPAddr.IP6_CRE.match(text)
-			if match:
-				return match
-			else:
-				return None
-
+		return match if match else None
 
+# An IPv4 compatible IPv6 to be reused
+_IPv6_v4COMPAT = IPAddr("::ffff:0:0", 96)
```

Comment:
I think it would not work on the top since class is not declared by then yet

On April 24, 2016 11:47:29 AM EDT, "Alexander Köppe" <notifications@github.com> wrote:
>>  
>> +# An IPv4 compatible IPv6 to be reused
>> +_IPv6_v4COMPAT = IPAddr("::ffff:0:0", 96)
>
>I think this kind of a constant looking like variable should be right
>at the top, just below the import statements.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/koeppea/fail2ban/pull/2/files/b9adbadb970043133f13448b50780d6de4ab5f37#r60846638